txcache/txCache.go

```diff
@@ -108,8 +108,14 @@ func (cache *TxCache) SelectTransactions(session SelectionSession, options commo
 	stopWatch := core.NewStopWatch()
 	stopWatch.Start("selection")
 
+	rootHash, err := session.GetRootHash()
+	if err != nil {
+		return nil, 0
```

Comment:
Since this should never happen in production, let's add an error log, similar to the one on line 104.